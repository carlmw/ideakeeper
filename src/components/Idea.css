:local(.idea) {
  padding: 4px;
  box-shadow: 0 0 2px lightgrey;
  display: flex;
  flex-direction: column;
  flex-flow: column;
}

:local(.field) {
  outline: none;
  border: 2px solid transparent;
  padding: 4px;
  width: 100%;
}

:local(.field):focus {
  border-color: lightgrey;
}

input:local(.field) {
  margin-right: 4px;
}

textarea:local(.field) {
  flex: 1;
  resize: none;
}

:local(.header) {
  display: flex;
  margin-bottom: 4px;
}

:local(.delete) {
  transition: opacity 0.16s;
  cursor: pointer;
  opacity: 0;
  outline: none;
  appearance: none;
  border: none;
  color: grey;
  font-size: 1em;
}

:local(.idea):hover :local(.delete) {
  opacity: 1;
}