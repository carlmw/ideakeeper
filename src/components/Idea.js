import React from 'react';
import PropTypes from 'prop-types';
import styles from './Idea.css';

const Idea = ({ title, body, onTitleChange, onBodyChange }) => {
  const onTitleBlur = (event) => onTitleChange(event.target.value);
  const onBodyBlur = (event) => onBodyChange(event.target.value);

  return (
    <div className={styles.idea}>
      <header>
        <input className={styles.field} type="text" defaultValue={title} onBlur={onTitleBlur} />
      </header>
      <textarea className={styles.field} maxLength={140} defaultValue={body} onBlur={onBodyBlur} />
    </div>
  );
};
Idea.propTypes = {
  title: PropTypes.string,
  body: PropTypes.string,
  onTitleChange: PropTypes.func.isRequired,
  onBodyChange: PropTypes.func.isRequired,
};

export default Idea;
