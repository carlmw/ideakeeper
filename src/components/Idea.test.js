import React from 'react';
import { shallow } from 'enzyme';
import Idea from './Idea';

describe('<Idea />', () => {
  const AN_IDEA = {
    id: '1',
    created_date: '2018-11-06T18:42:17.229Z',
    title: 'Chocolate teapot',
    body: 'A reusable teapot made from dark chocolate',
    onTitleChange: () => {},
    onBodyChange: () => {}
  };

  let result;
  beforeEach(() => {
    result = shallow(<Idea {...AN_IDEA} />);
  });

  it('renders an idea', () => {
    expect(result).toMatchSnapshot();
  });

  describe('when the title field blurs', () => {
    const NEW_TITLE = 'a puddle of chocolate';

    it('triggers onTitleChange', () => {
      const onTitleChange = jasmine.createSpy('onTitleChange');
      result.setProps({ onTitleChange });
      result.find('input').simulate('blur', { target: { value: NEW_TITLE } });

      expect(onTitleChange).toHaveBeenCalledWith(NEW_TITLE);
    });
  });

  describe('when the body field blurs', () => {
    const NEW_BODY = 'it melted';

    it('triggers onTitleChange', () => {
      const onBodyChange = jasmine.createSpy('onTitleChange');
      result.setProps({ onBodyChange });
      result.find('textarea').simulate('blur', { target: { value: NEW_BODY } });

      expect(onBodyChange).toHaveBeenCalledWith(NEW_BODY);
    });
  });
});
