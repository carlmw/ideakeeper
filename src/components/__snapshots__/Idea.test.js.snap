// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Idea /> renders an idea 1`] = `
<div>
  <header>
    <input
      defaultValue="Chocolate teapot"
      onBlur={[Function]}
      type="text"
    />
  </header>
  <textarea
    defaultValue="A reusable teapot made from dark chocolate"
    maxLength={140}
    onBlur={[Function]}
  />
</div>
`;
